### PR TITLE
Add single DynamoDB table state tracking migration option for AWS Kinesis inputs (KCL 3.5.0)

### PR DESCRIPTION
**Pull request by Nithin-Kasam**
## Description

-> Added a new advanced configuration option `kinesis_single_table_state_tracking` to the AWS Kinesis input, which stores all KCL state (leases, worker metrics, and coordinator state) in a single DynamoDB table instead of three separate tables.

- New `BooleanField` `kinesis_single_table_state_tracking` on `KinesisTransport` / `AWSInput`, defaulting to `true` for new inputs
- When enabled, graylog2-server/src/main/java/org/graylog/integrations/aws/transports/KinesisConsumer.java:62:0-223:1) sets `coordinatorConfig.migrateAllEntitiesToLeaseTable(true)`, triggering KCL's built-in one-way migration from the legacy multi-table layout
- [AWSInputCreateRequest]graylog2-server/graylog2-server/src/main/java/org/graylog/integrations/aws/resources/requests/AWSInputCreateRequest.java:0:0-19:40) / `AWSService` map the flag from the create request into the input configuration, defaulting to `false` when absent so existing inputs are not migrated unintentionally
- Web interface: new checkbox in the Kinesis setup Advanced Options, review step display, and payload mapping

## Motivation and Context

KCL 3.x creates three DynamoDB tables per application (leases, `CoordinatorState`, `WorkerMetricStats`), which increases cost and operational overhead. KCL 3.5.0 introduces a consolidated single-table layout with a one-way migration path for existing applications. This change exposes that capability as an input option.
 
**Note**: the migration is one-way and KCL applies a bake period (default 24h, min 1h) in the `PENDING` state before finalizing.

closes https://github.com/Graylog2/graylog-plugin-enterprise/issues/14774

## How Has This Been Tested?

- Manually tested end-to-end against AWS (ap-northeast-1):
  - Baseline with KCL 3.4.3: verified three separate tables are created
  - Upgraded to KCL 3.5.0 with flag disabled: verified Phase 1 (`TABLE_MIGRATION_STATUS_INIT` → `DEPLOYED`)
  - Enabled the flag on the existing input: verified Phase 2 (`DEPLOYED` → `PENDING` → `COMPLETE`) and that all entities (leases, coordinator state, worker metrics) were consolidated into the single lease table with `entityType` attributes.
  - Added unit testcases in both front end and backend.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have requested a documentation update.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by Nithin-Kasam** [comment]
Attaching testing Guide here 
[Kinesis Single Table Migration testing Guide.docx](https://github.com/user-attachments/files/30085180/Kinesis.Single.Table.Migration.testing.Guide.docx)

File: graylog2-server/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java [hunk 1]
```diff
@@ -202,6 +203,12 @@ public ConfigurationRequest combinedRequestedConfiguration() {
                     ConfigurationField.Optional.OPTIONAL,
                     NumberField.Attribute.ONLY_POSITIVE));
 
+            request.addField(new BooleanField(
+                    KinesisTransport.CK_KINESIS_SINGLE_TABLE_STATE_TRACKING,
+                    "Use single DynamoDB table for state tracking",
+                    true,
```
Comment (ryan-carroll-graylog) [inline graylog2-server/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java]: I noticed that a legacy input will have this checkbox enabled by default because the field is not present on the old config. If they go to edit the input for something unrelated, it will then flip to true and kick off this migration. 

Separately, I noticed we can uncheck this box even if it's already been enabled which may be confusing if this is truly irreversible.

Maybe we can talk it over a bit more with Rich and the team to fine tun e how we want this to behave since it's a pretty unique flow.

**Review by ryan-carroll-graylog — COMMENTED** [review]
This is looking good at first pass @Nithin-Kasam !

I see there are some merge conflicts, would you mind cleaning those up and then we can some more in depth review and testing.

File: graylog2-server/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java (hunk 1 above)
Comment (Nithin-Kasam) [reply graylog2-server/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java]: Hi @ryan-carroll-graylog,

For legacy inputs, the kinesis_single_table_state_tracking configuration variable is not present. As a result, in KinesisTransport (line 153), it defaults to false:

final boolean migrateToSingleTable =
    configuration.getBoolean(CK_KINESIS_SINGLE_TABLE_STATE_TRACKING, false);

This means Phase 1 of the migration starts automatically for legacy inputs. Later, when a user edits the input, the configuration is saved with the default value of true, which completes Phase 2 of the migration.

So the current behavior is:

Existing (legacy) inputs: default to false.
Newly created inputs: default to true.

Comment (ryan-carroll-graylog) [reply graylog2-server/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java]: Ok thanks for the clarification @Nithin-Kasam .

My main concern was if a user with a legacy input needs to edit an input (to just change batch size for example), but they're not ready to start phase 2. They may not notice that the checkbox is checked and save the input, kicking off the migration.

But you bring up a good point about wanting the box checked by default for new inputs.

I think it's fine to leave it as it is for now while we think through the edge case behavior and review/test the main logic though.

**Review by danotorrey — APPROVED** [review]
LGTM and tested successfully. Migration for input that existed (and received messages) before KCL `3.5` update tested successfully. Messages received successfully throughout the upgrade/migration process with no message loss.

**Review by ryan-carroll-graylog — APPROVED** [review]

**Review by kingzacko1 — APPROVED** [review]
